python/cli/prelude_cli/views/detect.py

```diff
@@ -172,8 +174,18 @@ def describe_activity(controller, days, view, tests, tags, endpoints, dos, statu
 
     raw = controller.describe_activity(view=view, filters=filters)
     report = Table()
+    build_tests = dict()
+
+    def get_test(test_id):
+        if show_test_id or not test_id:
+            return test_id
+        if not build_tests:
```

Comment:
this is a lot of work; let's just do this in 2 lines: 1) get all tests 2) add to dict

Reply:
What dict? Also I don't want to do that extra api call for views that don't need it which is the majority of them

Reply:
I'm trying to not duplicate code here, I would have the same block of code for all views returning test ids

Reply:
hmm; my issue here isn't about normalizing the code, it's the fact ``build.list_tests()`` already returns a dict of all tests, so there is no need to move a test from that collection to a new ``build_tests`` dict on demand. just look the value up in the dictionary.